### DISCUSSION TITLE
Optional list UploadFile

### DISCUSSION THREAD

**Discussion by OzodbekUSTECH** [fastapi/fastapi#10354] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [x] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.post('/reply/', response_model=IdResponseSchema)
async def create_reply_comment(
    media_group: list[UploadFile] = File(None),
    parent_comment_id: int = Form(...),
    user_id: int = Form(...),
    comment_text: str = Form(...),
):
    
    print(parent_comment_id)
```

### Description

it returns
{
  "detail": [
    {
      "type": "value_error",
      "loc": [
        "body",
        "media_group",
        0
      ],
      "msg": "Value error, Expected UploadFile, received: <class 'str'>",
      "input": "",
      "ctx": {
        "error": {}
      },
      "url": "https://errors.pydantic.dev/2.1/v/value_error"
    }
  ]
}

### Operating System

Windows

### FastAPI Version

0.103.2

### Pydantic Version

2.1.1

### Python Version

Python 3.11.4

**Comment by OzodbekUSTECH** [comment]
Updated pydantic and have the same error and the version of pydantic now is 2.4.2

**Comment by NeilBotelho** [comment]
Wrap your media_group type with Optional from typing, i.e.:
```python
from typing import Optional
@router.post('/reply/',response_model=IdResponseSchema)
async def create_reply_comment(
    media_group: Optional[list[UploadFile]]=File(None),
    parent_comment_id: int = Form(...),
    user_id: int = Form(...),
    comment_text: str = Form(...),
):
```

Also, I assume you're testing this in Swagger UI. When you make the request in swagger, untick the "Send empty value" option for the media_group. 
If you look at the curl command it generates, it adds a "media_group=" parameter which is causing the error. When you untick "Send empty value" it just doesn't send the parameter and FastAPI uses your default value.  
If you actually just dont send the parameter it should go through without any changes.

**Reply by OzodbekUSTECH** [reply]
But when i type with optional then i will get just string field instead of Uploadfile :( I have tried a lot of ways to avoid this problem... but coudnt find the solution
![image](https://github.com/tiangolo/fastapi/assets/126599107/0ac1cc8d-5fb2-4d8d-a82b-ef2b6513127e)

**Reply by NeilBotelho** [reply]
did you try unticking the send empty value checkbox?

**Reply by OzodbekUSTECH** [reply]
I guess its not good idea when i dont want to send any file

**Reply by tacticalowlman** [reply]
Sadly as soon as you apply Optional in code, swagger field looses option to upload file, no matter if "Send empty value" is checked or not :(

**Reply by adadgoff** [reply]
try this:
`file: UploadFile | str | None = File(None, media_type="image/*"),`

when `UploadFile is None`, it becomes `"null"` in string format.

Hope help you. Sorry for my english

**Comment by tacticalowlman** [comment]
Having the same problem

**Reply by adadgoff** [reply]
try this:
`file: UploadFile | str | None = File(None, media_type="image/*"),`

when `UploadFile is None`, it becomes `"null"` in string format.

Hope help you. Sorry for my english

**Reply by tacticalowlman** [reply]
Sadly any combination of proposed solution, removes ability to upload files from swagger.

**Reply by adadgoff** [reply]
Yeah, swagger dont work, but from frontend works good)

**Reply by jonas-debeuk** [reply]
Thanks @adadgoff that fixed the issue for me. Note I also added this condition to be safe in the router

```
from starlette.datastructures import UploadFile as StarletteUploadFile

...
    file: UploadFile | str | None = None
...
        if file and isinstance(file, (UploadFile, StarletteUploadFile)):
```

**Comment by pedroconceicao** [comment]
Having the same issue. Already shared it in a different [discussion](https://github.com/tiangolo/fastapi/discussions/10280#discussioncomment-9471549).
